src/resources/assets/js/mixins/largoContainer.vue

```diff
@@ -176,15 +194,39 @@ export default {
         }
     },
     methods: {
+        compileFilters(filters, union) {
+            let parameters = [];
+
+            filters.forEach(
+                (filter) => {
+                    if (!parameters[filter.filter]) {
+                        parameters[filter.filter] = [];
+                    }
+                    parameters[filter.filter].push(filter.value);
+                }
+            )
+            parameters['union'] = union;
+            return parameters;
+        },
         getAnnotations(label) {
             let promise1;
             let promise2;
 
+            //store in variables to avoid race conditions
+            let union = this.union;
+            let selectedFilters = this.selectedFilters;
+
             if (!this.annotationsCache.hasOwnProperty(label.id)) {
                 Vue.set(this.annotationsCache, label.id, []);
                 this.startLoading();
                 promise1 = this.queryAnnotations(label)
-                    .then((response) => this.gotAnnotations(label, response), handleErrorResponse)
+                    .then(
+                        (response) => this.gotAnnotations(label, response),
+                        handleErrorResponse
+                    )
+                    .then(a => Vue.set(this.annotationsCache, label.id, a))
+                    .then(this.loadFilters(label, selectedFilters, union))
```

Comment:
This is executed immediately. If you want to do it after the annotations were loaded, it has do be:

```suggestion
                    .then(() => this.loadFilters(label, selectedFilters, union))
```

But you can probably load the filters in parallel to the annotations, as it is currently done with the sorting sequence. To do this, remove the filter logic from this block of `promise1` and add a new block for `promise3` below like this:

```js
let promise3 = Promise.resolve();
if (selectedFilters.length > 0 && filtersNotInCache) {
    promise3 = this.loadFilters(...);
}

Vue.Promise.all([promise1, promise2, promise3]).finally(this.finishLoading);
```

You have to update `loadFilters()` to always return a promise then. Also see the three comments below.